src/main/java/killjoy/processing/Parser.java

```diff
@@ -0,0 +1,139 @@
+
+        try {
+            return Task.TaskType.valueOf(typeOfTask.toUpperCase());
+        } catch (IllegalArgumentException e) {
+            return null;
+        }
+    }
+
+    /**
+     * Parses the user input to determine the description of the task.
+     * @param input The user input.
+     * @return The description of the task.
+     */
+    public static String getDescription(String input) {
+        String[] inputSplitBySlash = input.split("/");
+        return inputSplitBySlash[0].replaceFirst("todo ", "")
+                .replaceFirst("deadline ", "").replaceFirst("event ", "");
+    }
+
+    /**
+     * Parses the user input to determine the by date of the task.
+     * @param input The user input.
+     * @return The by date of the task.
+     */
+    public static String getBy(String input) {
+        String[] inputSplitBySlash = input.split("/");
+        if (inputSplitBySlash.length < 2) {
+            return null;
+        }
+        return inputSplitBySlash[1].replaceFirst("by ", "");
+    }
+
+    /**
+     * Parses the user input to determine the from date of the task.
+     * @param input The user input.
+     * @return The from date of the task.
+     */
+    public static String getFrom(String input) {
+        String[] inputSplitBySlash = input.split("/");
+        if (inputSplitBySlash.length < 2) {
+            return null;
+        }
+        return inputSplitBySlash[1].replaceFirst("from ", "");
+    }
+
+    /**
+     * Parses the user input to determine the to date of the task.
+     * @param input The user input.
+     * @return The to date of the task.
+     */
+    public static String getTo(String input) {
+        String[] inputSplitBySlash = input.split("/");
+        if (inputSplitBySlash.length < 3) {
+            return null;
+        }
+        return inputSplitBySlash[2].replaceFirst("to ", "");
+    }
+
+    /**
+     * Parses the input to get index of task to mark, unmark or delete.
+     * @param input The user input.
+     * @return The index and type of action.
+     */
+    public static String[] parseMarkUnmarkDelete(String input) {
+        String[] inputAsList = input.split(" ");
+
+        if (inputAsList.length == 1) {
+            ui.displayEnterNumberMessage();
+            return null;
+        }
+        //Regex section referenced from https://jenkov.com/tutorials/java-regex/index.html
+        String regex = "^(mark|unmark|delete)(\\s)(\\d+)$";
+        Pattern pattern = Pattern.compile(regex);
+        Matcher matcher = pattern.matcher(input);
+
+        if (!matcher.find()) {
+            ui.displayInvalidCommandFormatMessage();
+            return null;
+        }
+        return new String[]{matcher.group(1), matcher.group(3)};
+    }
+
+    /**
+     * Parses the input to get the date and time.
+     * @param dateTime The date and time.
+     * @return The date and time.
+     */
+    public static LocalDateTime parseDateTime(String dateTime) {
```

Comment:
What "parse" means could be made clearer. Parse here returns a LocalDateTime, but parse below in parseTaskInfo returns a String[]. Maybe consider something like "parseStringToLocalDateTime".
```suggestion
    public static LocalDateTime parseStringToLocalDateTime(String stringDateTime) {
```